hack/installer/entrypoint.sh

```diff
@@ -16,15 +16,25 @@ EOF
 
 function setup_raw_device(){
   printf "Creating RAW device, this may take a moment..."
+
   if [[ -f ${TALOS_RAW} ]]; then
     rm ${TALOS_RAW}
   fi
+
+  # If defaults or a test file is specified, make a temp file
+  # to prevent collisions.
+  if [ ${TALOS_RAW} == "/out/talos.raw" ] || \
+     [ ${TALOS_RAW} == "/out/test.raw" ]; then
+    TALOS_RAW="$(mktemp ${TALOS_RAW}.XXXXXX)"
+    RENAME_TEMPFILE=true
+  fi
+
   dd if=/dev/zero of="${TALOS_RAW}" bs=1M count=0 seek=544
-  DEVICE=$(losetup -f)
+
   # NB: Since we use BLKRRPART to tell the kernel to re-read the partition
   # table, it is required to create a partitioned loop device. The BLKRRPART
   # command is meaningful only for partitionable devices.
-  losetup -P ${DEVICE} ${TALOS_RAW}
+  DEVICE=$(losetup --find --partscan --nooverlap --show ${TALOS_RAW})
```

Comment:
This is probably the only thing we want to change in this script. All the other changes seem related to things we need to do if the test image is not output and I think we should output the test image. Also not sure I see much value in the rename stuff.